### PR TITLE
Minor janitorial fixes and improvements

### PR DESCRIPTION
**Pull request by jonas**

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ class JavaInitializationSpec extends WordSpec with Matchers {
 
     "initializes the right field" in {
       akka.http.scaladsl.model.headers.LanguageRange.`*` // first we touch the scala one, it should force init the Java one
-      akka.http.javadsl.model.headers.LanguageRange.ALL // touching this one should not fail
+      val _ = akka.http.javadsl.model.headers.LanguageRange.ALL // touching this one should not fail
```
Comment (jlprat) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: What is the purpose of this "discarded" val?

Comment (jonas) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: From my understanding it's a regression test for some tricky code which ensures correct initialization of static fields in the Java API.

Blame points to this ticket: https://github.com/akka/akka/issues/20403

File: akka-http-core/src/main/scala/akka/http/impl/util/LogByteStringTools.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ private[akka] object LogByteStringTools {
 
   def logTlsInbound(name: String, maxBytes: Int = MaxBytesPrinted): Flow[SslTlsInbound, SslTlsInbound, NotUsed] =
     Flow[SslTlsInbound].log(name, {
-      case s @ SessionTruncated         ⇒ s
+      case s: SessionTruncated          ⇒ s
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/util/LogByteStringTools.scala]: ok

File: akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala (hunk 1 above)
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: I guess scalac would panic that there's a pure statement here and it takes no effect?
Wouldn't we need to do the same for the other values?

Comment (jonas) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: I'll validate tomorrow, but I agree that this is surprising.

Comment (jonas) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: OK, I've looked into this and managed to reproduce a NPE with this patch -https://gist.github.com/jonas/6ccec6a28a105c89e3793b9f410b71ef
based on the test case provided here https://github.com/akka/akka/issues/20403#issuecomment-215356663.

I will create a ticket with these findings.

Comment (jonas) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: Created #936

File: akka-parsing/src/main/scala/akka/macros/LogHelper.scala [hunk 3]
```diff
@@ -22,9 +22,9 @@ private[akka] trait LogHelper {
   /** Override to prefix every log message with a user-defined context string */
   protected def prefixString: String = ""
 
-  def debug(msg: String) = macro LogHelper.debugMacro
-  def info(msg: String) = macro LogHelper.infoMacro
-  def warning(msg: String) = macro LogHelper.warningMacro
+  def debug(msg: String): Unit = macro LogHelper.debugMacro
```
Comment (jrudolph) [inline akka-parsing/src/main/scala/akka/macros/LogHelper.scala]: Thanks for cleaning up after me.

**Pull request merged**